### PR TITLE
8270886: Crash in PhaseIdealLoop::verify_strip_mined_scheduling

### PR DESCRIPTION
**Pull request by rwestrel**
The crash occurs because an OuterStripMinedLoopEnd has only a single
projection. The root cause is that an IfNode has a single
projection. When the condition of that If constant folds, the IfNode
ends up with no use which causes it to be removed. The
OuterStripMinedLoopEnd projection is removed as a consequence.

When the If is constructed, it does have 2 projections. One of the
projections lead to an infinite loop. That infinite loop has a
safepoint node that's input to the Root node. When
remove_root_to_sfpts_edges() executes in Compile::Optimize(), that
edge is removed. But because PhaseRenumberLive runs next (a subclass
of PhaseRemoveUseless), the infinite loop is removed.

The fix is to simply move the call to remove_root_to_sfpts_edges()
after PhaseRemoveUseless.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8270886](https://bugs.openjdk.java.net/browse/JDK-8270886): Crash in PhaseIdealLoop::verify_strip_mined_scheduling

### Reviewers
 * [Vladimir Kozlov](https://openjdk.java.net/census#kvn) (@vnkozlov - **Reviewer**)
 * [Tobias Hartmann](https://openjdk.java.net/census#thartmann) (@TobiHartmann - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/4911/head:pull/4911` \
`$ git checkout pull/4911`

Update a local copy of the PR: \
`$ git checkout pull/4911` \
`$ git pull https://git.openjdk.java.net/jdk pull/4911/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 4911`

View PR using the GUI difftool: \
`$ git pr show -t 4911`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/4911.diff">https://git.openjdk.java.net/jdk/pull/4911.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/share/opto/compile.cpp [hunk 1]
```diff
@@ -2148,7 +2144,11 @@ void Compile::Optimize() {
     set_for_igvn(save_for_igvn);
   }
 
-  // Perform escape analysis
+  // Now that all inlining is over and no PhaseRemoveUseless will run, cut edge from root to loop
+  // safepoints
+  remove_root_to_sfpts_edges(igvn);
+
+    // Perform escape analysis
```
Comment (vnkozlov) [inline src/hotspot/share/opto/compile.cpp:2151]: Indent is wrong for this line. Please, fix it.

**Review by vnkozlov — COMMENTED** [review]
Looks good. What testing you did?

File: src/hotspot/share/opto/compile.cpp (hunk 1 above)
Comment (rwestrel) [reply src/hotspot/share/opto/compile.cpp]: fixed

**Comment by rwestrel** [comment]
> Looks good. What testing you did?

Thanks for the review.
hotspot/jtreg/compiler hotspot/jtreg/gc hotspot/jtreg/runtime jdk/java/lang jdk/java/util jtreg tests with -Xcomp -XX:CompileThreshold=100
hotspot/jtreg/compiler with no extra opton
some CTW testing.

**Review by vnkozlov — APPROVED** [review]
Good.

**Review by TobiHartmann — APPROVED** [review]
Looks good to me.

**Comment by rwestrel** [comment]
@vnkozlov @TobiHartmann thanks for the reviews.

**Pull request closed without merging**